api/src/main/java/io/minio/GetObjectResponse.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.minio;
+
+import java.io.IOException;
+import java.io.InputStream;
+import okhttp3.Headers;
+
+public class GetObjectResponse extends InputStream {
```

Comment:
Why not extend from `GenericResponse` and have just one instance variable for `InputStream body` ?

Reply:
`InputStream body` by `public InputStream body()` is already there.

We want `GetObjectResponse` needs to be an `InputStream` and have functionality of `GenericResponse`. As `GenericResponse` and `InputStream` are abstract classes, I cannot inherit both the classes, but inherit one class (`InputStream`) and have instance of another class (`GenericResponse`); if we do other way around; `InputStream` capability is lost.

Reply:
@balamurugana the `InputStream` capability won't be lost. We can get the `InputStream` by calling `body()` from the instance of `GetObjectResponse` to get the input stream.

Reply:
The idea here is to `GetObjectResponse` be an `InputStream` which is most common usage and it doesn't break backward compatibility.

Reply:
Backward compatibility with the previous return type of the method - understood :+1: 
In that case, may I suggest to remove all the wrapper methods that call the same method on the `response` object, and expect the client to call `response()` method to get the object of `BaseResponse` and then use it to get further details? We can also consider making the `GenericResponse` non-abstract and using it directly.